### ISSUE TITLE
Tag filter uses AND logic if several tags are selected - should use OR logic

### ISSUE THREAD

**Issue by silmaril42** [nextcloud/deck#2149]
**Describe the bug**
When filtering cards for tags and selecting several tags, only cards having all selected tags are shown (condition: card must have tagA AND tagB).

**To Reproduce**
1. have a board with at least 2 tags, tagA and tagB
2. assign tagA to one card
3. assing tagB to another card
4. assign both tagA and tagB to a third card
5. use the filter and select both tagA and tagB
6. notice that only the third card is shown

**Expected behavior**
I would expect to see all cards that have one of the selected tags (condition: card must have tagA OR tagB).
Of course, you could argue which of the two possibilities is the "correct" way to go - for me the current AND condition is of no use at all, while I could use OR conditions.

So I'm not sure if this is a bug report or a feature request - that depends on your original intentions while implementing that bit ;-)

If others have use cases for AND conditions, maybe the best way to go would be an option to chose which behaviour should be used?

**Side note**
The Nextcloud Deck app for Android is using OR conditions when selecting several tags in the filter, which feels (to me) like the natural behaviour.

**Nextcloud version:**
Nextcloud 19.0
Deck 1.0.5

**Comment by stefan-niedermann** [comment]
Related to https://github.com/nextcloud/deck/issues/1877

**Comment by juliushaertl** [comment]
> Unified search uses OR, but it make sense there as each result only have one type (contact, file, event...)

Isn't that up to the app depending on how you interpret the search term? I'd say this should always be AND by default. cc @jancborchardt

**Comment by skjnldsv** [comment]
> Isn't that up to the app depending on how you interpret the search term? I'd say this should always be AND by default. cc @jancborchardt

For the in-app filters, sure, for the type filters, it's on the Unified Search API to decide :)

**Comment by juliushaertl** [comment]
Ah I wasn't seeing the type context. I agree that those should indeed be or

**Comment by ImaCrea** [comment]
Hey! Thanks for Deck, we're using it everyday.

We're totally supporting "OR" instead of "AND" for a filter. We don't see a case where "AND" would be useful fur us, or maybe in very rare occasion. Meanwhile, having to filter one by one tags is not comfortable when we could just have a "OR" way of filtering.

I also checked how it's done on Trello and this is how filter is done, with a "OR" way. It's very probable they a/b tested it thoroughly and proved internally that's the most useful way to offer to their hundred of thousands of users.

**Comment by andreas1107** [comment]
+1 for the OR functionality

**Comment by elhananjair** [comment]
Hello, there is any progress on this issue?
I have created different tags named with the name of the month, and at the end of a quarter, I wanted to look back on each card, like what I have done in the past three months. 

Now, selecting three tags (November December, January) should list cards tagged with the specified names, but nothing will be shown when I choose the next tag in the filter.

**Comment by qguv** [comment]
> We're totally supporting "OR" instead of "AND" for a filter. We don't see a case where "AND" would be useful fur us, or maybe in very rare occasion. Meanwhile, having to filter one by one tags is not comfortable when we could just have a "OR" way of filtering.

The way I use tags, AND makes sense. Imagine one tag for tasks that need to be done at home and another tag for tasks that could earn money. Then, one could search for money-earning tasks that you can do at home by searching for both tags.

The use case of December, January, February is also very reasonable though! So I don't think we should have to choose between AND and OR—it should be possible to get either type of behavior. There's a nice way to do this… read on :)

Search engines typically assume "AND", e.g. the search `Player Piano Kurt Vonnegut` should give the book "Player Piano" by Kurt Vonnegut, not every work by Kurt Vonnegut as well as every work called "Player Piano") but also allow for OR with some special syntax, e.g. `"Kurt Vonnegut" OR "Player Piano"`. In fact, arbitrary combinations of AND and OR make a lot of sense: imagine looking for pet tasks at home: `Home AND (Cat OR Dog)`. Throw in NOT, and you have a very powerful search language indeed, allowing searches like `(Cat OR Dog) AND NOT Home`.

We know every complex combination of AND, OR, and NOT can be expressed as a [conjunction of multiple disjunctions](https://en.wikipedia.org/wiki/Conjunctive_normal_form) (or [vice versa](https://en.wikipedia.org/wiki/Disjunctive_normal_form)), so a syntax inspired by CNF could be helpful! Imagine you wanted to search for `December AND (Cat OR Dog)`; you could express this as two filters: one for `Cat OR Dog`, and one for `December`; which are ANDed together.

The interface might look something like this:

- click the button to add a filter, then tick the checkboxes for `Cat` and `Dog`, which would create a search for anything tagged Cat *or* Dog (as the issue suggests)
- then, click the filter button again to add a _second_ filter and select `December`
- the two filters, each representing a disjunction (an OR), would be ANDed together (a conjunction), and you would get any tasks tagged December which are also tagged either Cat or Dog.
- for bonus points: perhaps you could select tags for positive or negative inclusion, e.g. being able to add a search for things that do NOT have a particular tag. This might be implemented by a second click on the tag's checkbox to indicate a _negative_ condition. (A third click might uncheck the tag, making its inclusion or exclusion no longer relevant for this filter clause.)

**Comment by Flachzange** [comment]
I can only second on this. I am using deck for a couple of years now and everytime I stumble across the tag logic with implemented AND. It seems to be the only application in which tags are ANDed. For example in paperless-ngx there is option to chose between AND and OR:

<img width="406" height="398" alt="Image" src="https://github.com/user-attachments/assets/c8517c91-5245-4166-816f-e85a1dd4d366" />

**Comment by marcoambrosini** [comment]
I agree that for this particular use case we should show cards that has any of the tags, not all of them
